app/models/booking.rb

```diff
@@ -21,25 +21,31 @@ def self.upcoming(user)
   	# @user = User.check_user(user)
   	date_today = DateTime.now
     booking = nil
+    # Refactored queries to not use select for possible performance improvements
   	if user.admin?
-  		return Booking.order(created_at: :desc)
+  		# return Booking.order(:booking_date).select{ |booking| booking.booking_date > date_today}
+      return Booking.where('booking_date >= ?', date_today).order(:booking_date)
   	elsif user.presenter? # Add all booked then bids
-      return booking = Booking.where(chosen_presenter_id: user.id).select{ |booking| booking.booking_date > date_today}
+      # return Booking.with_deleted.where(chosen_presenter: user.presenter).order(:booking_date).select{ |booking| booking.booking_date >= date_today}
+      return Booking.with_deleted.where('booking_date >= ? AND chosen_presenter_id = ?', date_today, user.presenter).order(:booking_date)
     elsif user.customer?
-      return Booking.where(creator: user.customer).select{ |booking| booking.booking_date > date_today}
+      # return Booking.with_deleted.where(creator: user.customer).select{ |booking| booking.booking_date > date_today}
+      return Booking.with_deleted.where('booking_date >= ? AND creator_id = ?', date_today, user.customer).order(:booking_date)
   	end
   	return nil
   end
 
-  # Return completed bookings for a customer or presenter
-  # Return all completed bookings for an admin
+  # Return past bookings for a customer or presenter
+  # Return all past bookings for an admin
   def self.completed(user)
   	date_today = DateTime.now
     booking = nil
-  	if @user.presenter?
-  		return @user.presenter.bookings.with_deleted.order(created_at: :desc).select{ |booking| booking.booking_date < date_today}
-  	elsif @user.customer?
-  		return @user.customer.bookings.with_deleted.order(created_at: :desc).select{ |booking| booking.booking_date < date_today}
+  	if user.presenter
+  		return Booking.with_deleted..where('booking_date < ? OR deleted_at IS NOT NULL',date_today).where(chosen_presenter: user.presenter).order(:booking_date)
```

Comment:
Yep, although I think it was still working. Will fix it up tomorrow and pull in any new changes from dev as well.

> On 13 Jun 2016, at 7:15 PM, andrew <notifications@github.com> wrote:
> 
> In app/models/booking.rb:
> 
> >    def self.completed(user)
> >    	date_today = DateTime.now
> >      booking = nil
> > -  	if @user.presenter?
> > -  		return @user.presenter.bookings.with_deleted.order(created_at: :desc).select{ |booking| booking.booking_date < date_today}
> > -  	elsif @user.customer?
> > -  		return @user.customer.bookings.with_deleted.order(created_at: :desc).select{ |booking| booking.booking_date < date_today}
> > +  	if user.presenter
> > +  		return Booking.with_deleted..where('booking_date < ? OR deleted_at IS NOT NULL',date_today).where(chosen_presenter: user.presenter).order(:booking_date)
> Booking.with_deleted..where(... Typo?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>